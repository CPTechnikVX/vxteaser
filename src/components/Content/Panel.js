import React     from 'react';
import Constants from '../../utils/Constants';
import PropTypes from 'prop-types';

/**
 * Standard panel element
 */
export default class Panel extends React.PureComponent {
	static propTypes = {
		/** @ignore */
		config:   PropTypes.object,
		/** @ignore */
		children: PropTypes.node,
		width:    PropTypes.string,
	};

	render() {
		const {children, config, width} = this.props;
		const styleObj                  = {
			width: config.width ? config.width : width,
		};

		return <div className={Constants.ClassName.Panel} style={styleObj}>{children}</div>;
	}
}
