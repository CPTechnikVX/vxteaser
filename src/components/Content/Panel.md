```html
<Panel width="45%">
  .... some content hier
</Panel>
```
