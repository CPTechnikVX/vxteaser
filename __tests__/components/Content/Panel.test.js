import React     from 'react';
import {shallow} from 'enzyme';
import Panel     from '../../../src/components/Content/Panel';

describe('render panel', () => {
	test('with content', () => {
		const wrapper = shallow(
			<Panel width="55%" config={{width: '50%'}}>Panel content</Panel>
		);
		expect(wrapper).toMatchSnapshot();

		const style = wrapper.prop('style');
		expect(style.width).toBe('50%');
	});
});
