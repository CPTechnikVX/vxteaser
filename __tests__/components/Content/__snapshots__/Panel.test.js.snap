// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render panel with content 1`] = `
<div
  className="vxteaser-panel"
  style={
    Object {
      "width": "50%",
    }
  }
>
  Panel content
</div>
`;
